src/chains/ethereum/ethereum/src/api.ts

```diff
@@ -2782,6 +2782,176 @@ export default class EthereumApi implements Api {
       overrides
     );
   }
+
+  /**
+   * Retrieves past blocks
+   *
+   * @param blockCount - Requested range of blocks. Will return less than the requested range if not all blocks are available.
+   * @param newestBlock - Highest block of the requested range.
+   * @param rewardPercentiles - (Optional) A monotonically increasing list of percentile values. For each block in the requested range, the transactions will be sorted in ascending order by effective tip per gas and the coresponding effective tip for the percentile will be determined, accounting for gas consumed.
+   *
+   * @returns The fee history for the returned block range. This can be a subsection of the requested range if not all blocks are available. Includes the oldestBlock, baseFeePerGas, gasUsedRatio, and percentile rewards.
+   * @example for: blockCount: 2, newestBlock: 0x10, rewardPercentiles: [10, 25, 50]
+   * {
+   *   oldestBlock: "0x10",
+   *   reward:[["0x59682f00","0x2a7b749c11","0x2a7b749c11"]],
+   *   baseFeePerGas:["0x40c9459e7","0x420068cc1","0x420068cc1"]
+   *   gasUsedRatio: [0.5750394293729167, 0.50]
+   * };
+   *
+   */
+  @assertArgLength(2, 3)
+  async eth_feeHistory(
+    blockCount: QUANTITY,
+    newestBlock: QUANTITY | Ethereum.Tag,
+    rewardPercentiles?: number[]
+  ): Promise<Ethereum.FeeHistoryResult> {
+    let reward = [];
+    const baseFeePerGas = [];
+    const gasUsedRatio = [];
+    const blockchain = this.#blockchain;
+    const PAD_PRECISION = 16; // Matching geth/infura decimal precision formatting
+    const PRECISION = 10 ** PAD_PRECISION;
+    const PRECISION_BIG_INT = Quantity.from(PRECISION).toBigInt();
+    const PRECISION_BIG_INT_PERCENTILE = PRECISION_BIG_INT * 100n;
+    const ZERO_BIG_INT = Quantity.from(0).toBigInt();
+
+    let blocksToFetch = Quantity.toNumber(blockCount);
+    let newestBlockNumber = blockchain.blocks
+      .getEffectiveNumber(newestBlock)
+      .toNumber();
+    let oldestBlock;
+
+    let currentBlockNumber = newestBlockNumber;
+
+    // Because tags are supported in newestBlock, starting from the oldest block
+    // Would require hitting the db for the latest block twice. Once to calc to oldest
+    // and once to process the block.
+    while (blocksToFetch > 0 && currentBlockNumber >= 0) {
+      const currentBlock = await blockchain.blocks.get(currentBlockNumber);
+      oldestBlock = Quantity.from(currentBlockNumber).toString();
```

Comment:
`Quantity` _is_ the correct type to be returning from the api method (shouldn't convert it to string though, as `Quantity` is specifically designed to represent data returned from the api).

In order to avoid repeated conversions, I'd suggest:

```oldestBlock = currentBlockNumber;```

and:

```
return {
  oldestBlock: Quantity.from(oldestBlock),
 ...
};
```